### ISSUE TITLE
[Bug] Cellpose segmentation parameters are not handled properly

### ISSUE THREAD

**Issue by pakiessling** [gustaveroussy/sopa#89]
Hi Quentin,

I observed poor segmenation performance with Sopa compared to running Cellpose stand-alone.
I dug a little bit into it and noticed some things that cause this. All of this is for Cellpose 3.0.10.
For testing I am trying to segment the following image:

![grafik](https://github.com/gustaveroussy/sopa/assets/104848590/67c3f330-bb05-42d2-98ea-e96c80800bc5)

## Problem 1, custom cellpose model
In the line https://github.com/gustaveroussy/sopa/blob/e898b15786dcd2871e975032bd4cda02c2bf9a39/sopa/segmentation/methods.py#L42

model_type=model_type seems to overwrite pretrained_model=pretrained_model. That means if I initialize the cellpose methods like this:

```
method = sopa.segmentation.methods.cellpose_patch(
    diameter=200,
    channels=channels,
    flow_threshold=0.82,
    cellprob_threshold=-4.5,
    pretrained_model="my_beatiful_weights",
    min_size=0,
)
```

`my_beatiful_weights` are ignored and cyto3, the default value, is used instead.
This can be seen below, where the result is only improved when I remove the line model_type=model_type from methods.py :

![grafik](https://github.com/gustaveroussy/sopa/assets/104848590/bccd1a6e-260f-479b-a666-31325a9d2881)

## Problem 2 Application of blur, and CLAHE

It would be cool to be able to disable these processing steps. Passing `None` currently leads to an error with the skimage functions.
I would also like to pass a `kernel_size` to `exposure.equalize_adapthist`.

More importantly, blur and CLAHE seems to be applied to all image channels simultaneously, 
https://github.com/gustaveroussy/sopa/blob/e898b15786dcd2871e975032bd4cda02c2bf9a39/sopa/segmentation/stainings.py#L108

which leads to cellboundary images containing DAPI staining and the inverse. I dont think this is what we want. The contrast / blur should be applied to the channels separately to increase contrast to the background and not mix the separate channels. Illustration below, these are the images that are being passed to eval with either `blur` or `clahe` applied:

![grafik](https://github.com/gustaveroussy/sopa/assets/104848590/74c1757e-a50f-43e1-98f0-dbbbd3b825b2)

![grafik](https://github.com/gustaveroussy/sopa/assets/104848590/1d02613a-bba4-4e02-ab99-f479181fbe68)

Let me know if I can help out with a PR :)

## System
 - OS: Linux
 - Python version 3.10.13
<details>
<summary>Dependencies versions</summary>
<br>
Package                   Version
------------------------- ------------
aiobotocore               2.11.2
aiohttp                   3.9.1
aioitertools              0.11.0
aiosignal                 1.3.1
anndata                   0.10.3
appdirs                   1.4.4
array-api-compat          1.4
asciitree                 0.3.3
asttokens                 2.4.1
async-timeout             4.0.3
attrs                     23.2.0
botocore                  1.34.19
cellpose                  3.0.10
certifi                   2023.11.17
charset-normalizer        3.3.2
click                     8.1.7
click-plugins             1.1.1
cligj                     0.7.2
cloudpickle               3.0.0
colorcet                  3.0.1
comm                      0.2.2
ConfigArgParse            1.7
connection-pool           0.0.3
contourpy                 1.2.0
cycler                    0.12.1
dask                      2024.7.0
dask-expr                 1.1.7
dask-image                2023.8.1
datashader                0.16.0
datrie                    0.8.2
debugpy                   1.8.1
decorator                 5.1.1
distributed               2024.7.0
docutils                  0.20.1
dpath                     2.1.6
exceptiongroup            1.2.0
executing                 2.0.1
fasteners                 0.19
fastjsonschema            2.19.1
fastremap                 1.14.1
fcsparser                 0.2.8
filelock                  3.13.1
fiona                     1.9.5
flake8                    7.0.0
fonttools                 4.47.0
frozenlist                1.4.1
fsspec                    2023.6.0
geopandas                 0.14.1
gitdb                     4.0.11
GitPython                 3.1.40
h5py                      3.10.0
humanfriendly             10.0
idna                      3.6
imagecodecs               2024.1.1
imageio                   2.33.1
importlib-metadata        7.0.1
ipykernel                 6.29.3
ipython                   8.22.2
jedi                      0.19.1
Jinja2                    3.1.2
jmespath                  1.0.1
joblib                    1.3.2
jsonschema                4.20.0
jsonschema-specifications 2023.12.1
jupyter_client            8.6.1
jupyter_core              5.6.1
kiwisolver                1.4.5
lamin_utils               0.13.0
lazy_loader               0.3
legacy-api-wrap           1.4
llvmlite                  0.41.1
locket                    1.0.0
loompy                    3.0.7
markdown-it-py            3.0.0
MarkupSafe                2.1.3
matplotlib                3.8.2
matplotlib-inline         0.1.6
matplotlib-scalebar       0.8.1
mccabe                    0.7.0
mdurl                     0.1.2
mpmath                    1.3.0
msgpack                   1.0.7
multidict                 6.0.4
multipledispatch          1.0.0
multiscale_spatial_image  1.0.0
natsort                   8.4.0
nbformat                  5.9.2
nest_asyncio              1.6.0
networkx                  3.2.1
numba                     0.58.1
numcodecs                 0.12.1
numpy                     1.23.4
numpy-groupies            0.10.2
nvidia-cublas-cu12        12.1.3.1
nvidia-cuda-cupti-cu12    12.1.105
nvidia-cuda-nvrtc-cu12    12.1.105
nvidia-cuda-runtime-cu12  12.1.105
nvidia-cudnn-cu12         8.9.2.26
nvidia-cufft-cu12         11.0.2.54
nvidia-curand-cu12        10.3.2.106
nvidia-cusolver-cu12      11.4.5.107
nvidia-cusparse-cu12      12.1.0.106
nvidia-nccl-cu12          2.18.1
nvidia-nvjitlink-cu12     12.3.101
nvidia-nvtx-cu12          12.1.105
ome-zarr                  0.9.0
opencv-python             4.9.0.80
opencv-python-headless    4.9.0.80
packaging                 23.2
pandas                    2.1.4
param                     2.0.1
parso                     0.8.3
partd                     1.4.1
patsy                     0.5.5
pexpect                   4.9.0
pickleshare               0.7.5
pillow                    10.2.0
PIMS                      0.6.1
pip                       23.3.2
plac                      1.4.2
platformdirs              4.1.0
pooch                     1.8.1
prompt-toolkit            3.0.43
psutil                    5.9.7
ptyprocess                0.7.0
PuLP                      2.7.0
pure-eval                 0.2.2
pyarrow                   14.0.2
pycodestyle               2.11.1
pyct                      0.5.0
pyflakes                  3.2.0
pygeos                    0.14
Pygments                  2.17.2
pynndescent               0.5.11
pyparsing                 3.1.1
pyproj                    3.6.1
python-dateutil           2.8.2
pytz                      2023.3.post1
PyYAML                    6.0.1
pyzmq                     25.1.2
readfcs                   1.1.7
referencing               0.32.0
requests                  2.31.0
reretry                   0.11.8
rich                      13.7.0
roifile                   2023.8.30
rpds-py                   0.16.2
s3fs                      0.6.0
scanpy                    1.10.2
scikit-image              0.22.0
scikit-learn              1.3.2
scipy                     1.11.4
seaborn                   0.13.1
session-info              1.0.0
setuptools                69.5.1
shapely                   2.0.2
six                       1.16.0
slicerator                1.1.0
smart-open                6.4.0
smmap                     5.0.1
snakemake                 7.32.4
sopa                      1.1.1
sortedcontainers          2.4.0
spatial_image             1.1.0
spatialdata               0.2.1
spatialdata-io            0.1.3.post0
spatialdata-plot          0.2.3.post0
stack-data                0.6.2
statsmodels               0.14.1
stdlib-list               0.10.0
stopit                    1.1.2
sympy                     1.12
tabulate                  0.9.0
tangram-sc                1.0.4
tblib                     3.0.0
threadpoolctl             3.2.0
throttler                 1.2.2
tifffile                  2023.12.9
toml                      0.10.2
toolz                     0.12.0
toposort                  1.10
torch                     2.1.2
tornado                   6.4
tqdm                      4.66.1
traitlets                 5.14.1
triton                    2.1.0
typer                     0.9.0
typing_extensions         4.10.0
tzdata                    2023.4
umap-learn                0.5.5
urllib3                   2.0.7
wcwidth                   0.2.13
wheel                     0.42.0
wrapt                     1.16.0
xarray                    2023.12.0
xarray-dataclasses        1.7.0
xarray-datatree           0.0.13
xarray-schema             0.0.3
xarray-spatial            0.3.7
yarl                      1.9.4
yte                       1.5.4
zarr                      2.16.1
zict                      3.0.0
zipp                      3.17.0
</details>

**Comment by quentinblampey** [comment]
Hello @pakiessling, thanks for digging into this, it seems to be a major fix! I'll be happy to review your PR 🙂

(A few comments regarding the PR: please merge your changes into the `dev` branch. I also recommend to use [`pre-commit`](https://pre-commit.com/), else the CI will likely fail)
